shared/structures/src/helpers/EventPermissionChecker.ts

```diff
@@ -0,0 +1,261 @@
+            getOrganization: (id: string) => Promise<O>;
+            userPermissions: UserPermissions | null;
+            platform: Platform;
+        },
+    ): Promise<O | null> {
+        if (!userPermissions) {
+            throw new SimpleError({
+                code: 'permission_denied',
+                message:
+                    'Je hebt geen toegangsrechten om een activiteit te beheren.',
+                statusCode: 403,
+            });
+        }
+
+        if (event.organizationId !== null) {
+            let organization: O;
+
+            try {
+                organization = await getOrganization(event.organizationId);
+            }
+            catch (error) {
+                throw new SimpleError({
+                    code: 'not_found',
+                    message: 'Event not found',
+                    human: 'De activiteit werd niet gevonden',
+                });
+            }
+
+            this.tryAdminOrganizationEvent(
+                event,
+                organization,
+                platform,
+                userPermissions,
+            );
+            return organization;
+        }
+
+        this.tryAdminNationalOrRegionalEvent(event, userPermissions, platform);
+        return null;
+    }
+
+    /**
+     * Will throw error if not allowed to edit/add/delete this event
+     * @param event
+     * @throws error if not allowed to write this event
+     */
+    static checkEventAccess<O extends OrganizationForPermissionCalculation>(
+        event: EventDataForPermission,
+        {
+            organization,
+            userPermissions,
+            platform,
+        }: {
+            organization: O | null;
+            userPermissions: UserPermissions | null;
+            platform: Platform;
+        },
+    ): void {
+        if (!userPermissions) {
+            throw new SimpleError({
+                code: 'permission_denied',
+                message:
+                    'Je hebt geen toegangsrechten om een activiteit te beheren.',
+                statusCode: 403,
+            });
+        }
+
+        if (event.organizationId !== null) {
+            if (organization === null) {
+                throw new Error('Organization not specified.');
+            }
+            this.tryAdminOrganizationEvent(
+                event,
+                organization,
+                platform,
+                userPermissions,
+            );
+        }
+        else {
+            this.tryAdminNationalOrRegionalEvent(event, userPermissions, platform);
+        }
+    }
+
+    static canAdminEvent<O extends OrganizationForPermissionCalculation>(
+        event: EventDataForPermission,
+        options: {
+            organization: O | null;
+            userPermissions: UserPermissions | null;
+            platform: Platform;
+        },
+    ): boolean {
+        try {
+            this.checkEventAccess(event, options);
+        }
+        catch (error) {
+            console.error(error);
```

Comment:
Fatal errors en niet-simple-errors moeten opnieuw gethrowd worden en niet worden weggegooid

Reply:
Ik bedoel dat dit gevaarlijk is, je moet hier enkel de verwachte errors opvangen, en de andere errors moet je throwen, anders ga je mogelijks belangrijke errors weggooien en zie je geen errors opduiken voor bugs